### PR TITLE
Configuration is now handled with Cobra and Viper.

### LINKED ISSUE #39: Create a better configuration system based on yaml files
We can try do golang libraries to do this once we agree on which one to use.

### PR DESCRIPTION
**Pull request by stephenwithav**
This commit also removes support for proxy_mode, as mentioned here:

https://github.com/Beanstalkg/beanstalkg/issues/39#issuecomment-306454795

Closes #39.

### CODE REVIEW

**Comment by vimukthi-git** [comment]
@stephenwithav looks good please add the new libs to godeps.

File: root.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	listener, err := net.ListenTCP("tcp", tcpAddr)
+	checkError(err)
+	stop := make(chan bool)
+
+	tubeRegister := make(chan architecture.Command)
+	// use this tube to send the channels for each individual tube to the clients when the do 'use' command
+	useTubeConnectionReceiver := make(chan chan architecture.Command)
+	watchedTubeConnectionsReceiver := make(chan chan architecture.Command)
+	operation.NewTubeRegister(tubeRegister, useTubeConnectionReceiver, watchedTubeConnectionsReceiver, stop)
+	log.Info("BEANSTALKG listening on: ", port)
```
Comment (vimukthi-git) [inline root.go]: Lets initiate the `log` explicitly in each file we use it.

File: root.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package main
```
Comment (vimukthi-git) [inline root.go]: shall we rename this file to `server.go`? `root.go` doesn't sound nice to me.

Comment (stephenwithav) [inline root.go]: Hah!  Gotcha.  `cobra` suggests `root`, but I understand.

Comment (vimukthi-git) [inline root.go]: Ok if `cobra` has a convention like that then we can go with that no need to change.

File: root.go (hunk 1 above)
Comment (stephenwithav) [inline root.go]: The best practices you linked in #66 recommends embedding the logger in a struct so it's not a free variable.  That would fix the logging issues we discussed earlier.

Would you object to me handling all logging issues in a separate PR?

**Comment by stephenwithav** [comment]
@vimukthi-git adding `viper` and `cobra` with godep adds a ton of dependencies.  Is there a way to omit their dependencies?

File: root.go (hunk 1 above)
Comment (vimukthi-git) [inline root.go]: no worries, I will make an update for logging following the guide.Thanks

**Comment by vimukthi-git** [comment]
@stephenwithav I honestly don't know if it can be avoided. lets do some research.

**Comment by stephenwithav** [comment]
Should I go ahead and push what

**Comment by vimukthi-git** [comment]
you can push, then I can see the diff

**Comment by stephenwithav** [comment]
Added.

File: Godeps/Godeps.json [hunk 3]
```diff
@@ -1,17 +1,150 @@
...
+		{
+			"ImportPath": "github.com/stretchr/objx",
+			"Rev": "1a9d0bb9f541897e62256577b352fdbc1fb4fd94"
+		},
+		{
+			"ImportPath": "github.com/stretchr/testify/assert",
+			"Comment": "v1.1.3-19-gd77da35",
+			"Rev": "d77da356e56a7428ad25149ca77381849a6a5232"
+		},
+		{
```
Comment (vimukthi-git) [inline Godeps/Godeps.json]: can you check why test related dependancies are getting added?

**Comment by vimukthi-git** [comment]
@stephenwithav You are correct, this introduces too many dependancies just for something as simple as config params. Lets find a simpler way. As you may understand more dependancies means more things that can go wrong.

**Comment by vimukthi-git** [comment]
@stephenwithav shall we close this?

**Comment by stephenwithav** [comment]
@vimukthi-git Yes.

**Pull request closed without merging**